sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_value.hpp

```diff
@@ -70,146 +76,668 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
+     *
+     * @remarks This is an internal accessor and should never be used by code outside the AMQP
+     * implementation.
+     *
+     * @param value - the AMQP array value to capture.
+     */
+    AmqpList(AMQP_VALUE_DATA_TAG* const value);
+    friend std::ostream& operator<<(std::ostream& os, AmqpList const& value);
+  };
 
-  protected:
-    AMQP_VALUE_DATA_TAG* m_value;
+  /** @brief An AMQP binary value, a sequence of octets
+   *
+   * Defined in [AMQP Core Types
+   * section 1.6.19](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-binary).
+   *
+   */
+  class AmqpBinaryData
+      : public _detail::AmqpCollectionBase<std::vector<std::uint8_t>, AmqpBinaryData> {
+  public:
+    AmqpBinaryData() : AmqpCollectionBase(){};
+    /** @brief Construct a new AmqpList object with an initializer list. */
+    AmqpBinaryData(initializer_type const& values) : AmqpCollectionBase(values){};
+    AmqpBinaryData(std::vector<std::uint8_t> const& values) : AmqpCollectionBase(values){};
+
+    /** @brief Construct a new AmqpBinaryData object from an existing uAMQP AMQP_VALUE item
+     *
+     * @remarks Note that this does NOT capture the passed in AMQP_VALUE object, the caller is
+     * responsible for freeing that object.
+     *
+     * @remarks This is an internal accessor and should never be used by code outside the AMQP
+     * implementation.
+     *
+     * @param value - the AMQP "binary" value to capture.
+     */
+    AmqpBinaryData(AMQP_VALUE_DATA_TAG* const value);
+    friend std::ostream& operator<<(std::ostream& os, AmqpBinaryData const& value);
+  };
+
+  class AmqpSymbol : public _detail::AmqpCollectionBase<std::string, AmqpSymbol> {
+  public:
+    AmqpSymbol() : AmqpCollectionBase(){};
+    /** @brief Construct a new AmqpSymbol object with an initializer list. */
+    AmqpSymbol(std::string const& values) : AmqpCollectionBase(values){};
+    AmqpSymbol(initializer_type const& initializer) : AmqpCollectionBase(initializer) {}
+    AmqpSymbol(const char* const values) : AmqpCollectionBase(values){};
+
+    /** @brief Construct a new AmqpSymbol object from an existing uAMQP AMQP_VALUE item
+     *
+     * @remarks Note that this does NOT capture the passed in AMQP_VALUE object, the caller is
+     * responsible for freeing that object.
+     *
+     * @remarks This is an internal accessor and should never be used by code outside the AMQP
+     * implementation.
+     *
+     * @param value - the AMQP "binary" value to capture.
+     */
+    AmqpSymbol(AMQP_VALUE_DATA_TAG* const value);
+
+    friend std::ostream& operator<<(std::ostream& os, AmqpSymbol const& value);
+
+    bool operator==(AmqpSymbol const& that) const { return m_value == that.m_value; }
+    bool operator==(const decltype(m_value)::value_type* const that) const
+    {
+      return m_value == that;
+    }
   };
 
-  /** @brief An AmqpMap represents an AMQP "map" type. which maps a key to a value.
+  class AmqpTimestamp {
```

Comment:
BTW, I think it is good to just review all class and struct declarations in core-amqp for having `final` and depending on that having/not having virtual/protected destructor.
```suggestion
  class AmqpTimestamp final {
```